library/catalog/static/js/registrationForm.js

```diff
@@ -0,0 +1,128 @@
+const emailPattern = /^(\w|\.|_|-)+[@](\w|_|-|\.)+[.]\w{2,3}$/;
+const passPattern = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
+
+let form = document.querySelector('.needs-validation');
+let inputsList = document.querySelectorAll('input.form-control');
+
+let firstname = form.elements.namedItem('firstname');
+let lastname = form.elements.namedItem('lastname');
+let email = form.elements.namedItem('email');
+let login = form.elements.namedItem('login');
+let password = form.elements.namedItem('password');
+let confirmPassword = form.elements.namedItem('confirm_password');
+let submitButton = form.elements.namedItem('submitButton');
+
+firstname.addEventListener('input', validate);
+lastname.addEventListener('input', validate);
+email.addEventListener('input', validate);
+password.addEventListener('input', validate);
+confirmPassword.addEventListener('input', validate);
+login.addEventListener('input', validate);
+
+
+function validate(event) {
```

Comment:
This function is too complicated. From my point of view it will be good to have separate validation functions for each field.